server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -485,11 +486,11 @@ private void recoverFromTranslogInternal(TranslogRecoveryRunner translogRecovery
         // note: if opsRecovered == 0 and we have older translogs it means they are corrupted or 0 length.
         assert pendingTranslogRecovery.get() : "translogRecovery is not pending but should be";
         pendingTranslogRecovery.set(false); // we are good - now we can commit
+        logger.trace(() -> new ParameterizedMessage(
```

Comment:
Adjusted this because it's called more often now and `translog.currentFileGeneration()` actually does lock and create a `ReleaseableLock` instance